packages/teams-js/src/internal/communication.ts

```diff
@@ -807,6 +812,7 @@ function handleIncomingMessageFromParent(evt: DOMMessageEvent): void {
     if (callbackId) {
       const callback = CommunicationPrivate.callbacks.get(callbackId);
       logger('Received a response from parent for message %s', callbackId.toString());
+      HostToAppMessageDelayTelemetry.handlePerformanceMetrics(callbackId, message, logger);
```

Comment:
From the screenshot you have attached, one observation is that the themeChange messageDelay(from 27-39s) is much longer than getContext(in the range of 2.3s-9s). Not sure if it reflects the reality that themeChange MessageDelay is slower than getContext, or it includes some random variation which is common issue with manual testing

Also, If we look at the handlePerformanceMetrics and handleOneWayPerformanceMetrics method, both of them use getCurrentTimestamp as the ending timestamp. Do you think it makes sense to record a common ending time in handleIncomingMessageFromParent(for example, after line 805), then pass that endtime to handlePerformanceMetrics and handleOneWayPerformanceMetrics to calculate the MessageDelay later?

Reply:
The timestamp is in milliseconds hehe... So, the difference is not as big as it seems. It could be a lot of reasons of why this occurs, but it would be good to see the data in a production environment to be totally sure.

I find it reasonable to use the same ending time, the difference tho would be of some small milliseconds at most.

Reply:
Pushed in latest commit 😄